lib/rules/valid-v-model.js

```diff
@@ -49,7 +49,12 @@ function isOptionalChainingMemberExpression(node) {
  * @returns {boolean} `true` if the node can be LHS.
  */
 function isLhs(node) {
-  return node.type === 'Identifier' || node.type === 'MemberExpression'
+  return (
+    node.type === 'Identifier' ||
+    node.type === 'MemberExpression' ||
+    node.type === 'TSAsExpression' ||
+    node.type === 'TSNonNullExpression'
```

Comment:
Hmm. I think we need to check if the actual expression in LHS. For example, `foo()!` cannot be used in `v-model`, but I think it will be allowed with this change.

Reply:
Thanks for catching this! I missed it somehow...